### ISSUE TITLE
User can log out and only view pages when logged in

### ISSUE THREAD

**Issue by hpjaj** [oddballio/magnifier#63]
## Background
Authenticated users can access pages, guests cannot.
 
## Definition of Done
 
- [ ] All pages can only be accessed by logged in users
- [ ] There is a logout button that logs a user out

**Comment by RobAWilkinson** [comment]
@hpjaj I can take care of this, theres some new approaches in react-router 4 I believe. Could be a fun one for you to tackle too.

**Comment by hpjaj** [comment]
@RobAWilkinson  Yeah man, it’s all you!  I believe the sessions#destroy method and current users presence are options for you, too.

**Comment by RobAWilkinson** [comment]
True, but all the routing is done from the front end, right now user can see all links and pages, even when logged out.
Session.destroy will just make `current_user` nil, but won't redirect react-router or anything

**Comment by hpjaj** [comment]
Yeah, good call.  I was thinking an option would be to check for the presence of the current user at either the controller or `pages#index` view level, but that would mean we'd have to redirect users to a rails view page, instead of our react login page, and that's no bueno.

**Comment by hpjaj** [comment]
@RobAWilkinson - Is this inline with what you're thinking?

https://youtu.be/ojYbcon588A

**Comment by RobAWilkinson** [comment]
@hpjaj yes! Exactly, tyler mcginnis is gold

**Comment by hpjaj** [comment]
Sweet!  I can take this one if you'd like?
